### PR TITLE
[SERV-1095] Add release and nightly GA workflows

### PR DESCRIPTION
**Pull request by angelahuqing**
- Both workflows release images to Dockerhub
- Update names from service-template to pairtree-service

### CODE REVIEW

File: .github/workflows/nightly.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+name: Build and Publish Docker Image for Snapshot
+
+env:
+    GO_VERSION: '1.22.2'
+    SERVICE_NAME: 'pairtree-service'
+    ALPINE_VERSION: '3.19'
```
Comment (ksclarke) [inline .github/workflows/nightly.yml:6]: Alpine is up to 3.20.0 now.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
+name: Build and Publish Docker Image
+
+env:
+    GO_VERSION: '1.22.2'
+    SERVICE_NAME: 'pairtree-service'
+    ALPINE_VERSION: '3.19'
```
Comment (ksclarke) [inline .github/workflows/release.yml:6]: Alpine is up to 3.20.0 now.

File: Dockerfile [hunk 3]
```diff
@@ -2,9 +2,9 @@
 ## A Dockefile for UCLA Library's GoLang microservices.
 ##
 
-ARG GO_VERSION=1.22.1
+ARG GO_VERSION=1.22.2
 ARG ALPINE_VERSION=3.19
```
Comment (ksclarke) [inline Dockerfile:6]: 3.20.0

File: testcontainer_test.go [hunk 4]
```diff
@@ -64,7 +64,7 @@ func TestApp(t *testing.T) {
 	// Any requeset body will work since POST requests are not currently allowed
 	requestBody := []byte(`{"key": "value"}`)
 
-	resp, err = client.Post("http://" + host + ":" + strconv.Itoa(port.Int()) + "/", "application/json", bytes.NewBuffer(requestBody))
+	resp, err = client.Post("http://"+host+":"+strconv.Itoa(port.Int())+"/", "application/json", bytes.NewBuffer(requestBody))
```
Comment (ksclarke) [inline testcontainer_test.go:67]: Same question as the other PR. Is there a reason for getting rid of the spaces (gofmt or something?) It seems to be more readable with them, to me.

Comment (angelahuqing) [reply testcontainer_test.go:67]: VSCode removes the spaces for some reason as a part of their code-saving process. I agree the spaces make more sense, and we should keep them.

**Review by DRickard — APPROVED** [review]

**Review by ksclarke — APPROVED** [review]

**Pull request merged**